soccer/gameplay/evaluation/ball.py

```diff
@@ -86,15 +67,25 @@ def opponent_with_ball():
 
 # based on face angle and distance, determines if the robot has the ball
 def robot_has_ball(robot):
-    def angle_btw_three_pts(a, b, vertex):
-        return (a-vertex).angle_between(b-vertex)
-
-    angle = robot.angle
-    theta = angle_btw_three_pts(robot.pos + robocup.Point(
-        math.cos(angle), math.sin(angle)), main.ball().pos, robot.pos)
-    max_radius = constants.Robot.Radius * (2.0 + 2.0 * math.cos(theta))
-
-    return robot.pos.near_point(main.ball().pos, max_radius)
+    mouth_half_angle = 15*math.pi/180 # Angle from front
+    max_dist_from_mouth = 0.5 * constants.Robot.Radius
```

Comment:
This needs to be tuned a little more. I think max distance from mouth needs to actually be like `1.2 * Robot.Radius` but it's one of those things that just needs to be played with

Reply:
Ideally we should really just be able to use Robot::hasBall
Depends on Ball sense i guess. hasBall already has a _lostBallDuration delay.

Reply:
* I mean this is used for oponent robots too I guess. But for our robots we should really have a bit better idea than just this. Since we sort of use this randomly in alot of different places.

Reply:
It might be a good idea to just move this to the base `Robot` class and overload it with ball sense for our robots.